src/sqlfluff/core/rules/base.py

```diff
@@ -802,28 +802,32 @@ def _choose_anchor_segment(
 
         anchor: BaseSegment = segment
         child: BaseSegment = segment
-        for seg in context.parent_stack[0].path_to(segment)[1:-1][::-1]:
```

Comment:
Review this bit hiding whitespace:

<img width="364" alt="image" src="https://user-images.githubusercontent.com/10931297/157355093-b135bdfc-528f-4790-b4da-22cf4d6bbabc.png">